assets/app/scripts/controllers/createRoute.js

```diff
@@ -31,15 +31,33 @@ angular.module('openshiftConsole')
             return;
           }
 
-          $scope.routing.portOptions = _.map(service.spec.ports, function(portMapping) {
-            return {
-              containerPort: portMapping.targetPort,
-              protocol: portMapping.protocol
+          var unnamedServicePort;
+          if (service.spec.ports.length === 1 && !service.spec.ports[0].name) {
+            unnamedServicePort = true;
+            $scope.alerts['unnamed-service-port'] = {
+              type: "warning",
+              message: "Service " + service.metadata.name + " has a port with no name.",
+              details: "Unnamed service ports cannot be used as the target port for the route. " +
```

Comment:
Trying to think of the clearest way to word this...
"A route cannot specifically target an unnamed service port. If more service ports are added to this service in the future, the route will also direct traffic to them."?